Premium Content Block: show premium content after subscribe

rcrdortiz opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->
<!-- Would you like this feature to be tested by Beta testers?
Please add testing instructions to projects/plugins/jetpack/to-test.md in a new commit as part of your PR. -->
<!-- a12s: If you have an expected version that you're aiming for the PR to add, please use the Milestone field to communicate it. If you leave it blank, that indicates there isn't a preference. -->

Fixes [#23135](https://github.com/Automattic/jetpack/issues/23135) & 458-gh-Automattic/payments-shilling

#### Changes proposed in this Pull Request:
<!--- Explain what functional changes your PR includes -->
* Now when a user finishes the subscription process he is able to see the premium contents.

#### Jetpack product discussion
<!-- If you're an Automattician, include a shortlink to the p2 discussion with Jetpack Product here. -->
<!-- Make sure any changes to existing products have been discussed and agreed upon -->

#### Does this pull request change what data or activity we track or use?
<!--- If so, please add the "[Status] Needs Privacy Updates" label and explain what changes there are. -->
<!--- Check existing Jetpack support documents for a preview of the information we need. -->

#### Testing instructions:
<!-- If you were reviewing this PR, how would you like the instructions to be presented? -->
<!-- Please include detailed testing steps, explaining how to test your change. -->
<!-- Bear in mind that context you working on is not obvious for everyone.  -->
<!-- Adding "simple" configuration steps will help reviewers to get to your PR as quickly as possible. -->
<!-- "Before / After" screenshots can also be very helpful when the change is visual. -->

##### Testing premium content can't be viewed.
* Sandbox the store (Recommended).
* View as a user a post that has a premium content block (make sure you aren't logged in).
<img width="1323" alt="Captura de pantalla 2022-03-07 a las 13 42 31" src="https://user-images.githubusercontent.com/1989914/157039991-165c3b78-276b-493d-ad1e-e894a2a2fbdb.png">
* Click on the subscribe button and complete the subscription process.

<img width="1294" alt="Captura de pantalla 2022-03-07 a las 13 42 41" src="https://user-images.githubusercontent.com/1989914/157040018-b84db5f2-973d-45b5-a51c-a8acf407dc18.png">
* Close the subscription window, you should see the premium content.
<img width="1317" alt="Captura de pantalla 2022-03-07 a las 13 47 04" src="https://user-images.githubusercontent.com/1989914/157040055-d63ba9bd-1185-4b99-94b6-e5a8de207e47.png">

##### Testing the login button continues to show after logging in
* The best way to test this scenario is to add an invalid token as a parameter.
* Get the post URL and token=invalid
* Open the URL
* You should not see the login button and see the subscribe button.
* You should not see the premium content.
<img width="1289" alt="Captura de pantalla 2022-03-07 a las 13 50 05" src="https://user-images.githubusercontent.com/1989914/157040536-97419881-9f5d-45d9-b706-acfd3cdaa8fe.png">

rcrdortiz committed:
[not verified] Now after subscribing to premium content the page will reload. When a user has a premium content token we will not show the login button

rcrdortiz committed:
Now after subscribing to premium content the page will reload. When a user has a premium content token we will not show the login button

rcrdortiz committed:
Added changelog

rcrdortiz committed:
[not verified] Added changelog

rcrdortiz committed:
Merge branch 'master' into fix/show-premium-content-after-subscribe

thingalon committed:
Update changelog for Boost 1.4.1 (#23324)

* Update changelog for Boost 1.4.1

* Fix versioning hiccups

* Fix lockfiles

Co-authored-by: Mark George <thingalon@gmail.com>
Co-authored-by: Adnan Haque <adnan007.id@gmail.com>

thingalon committed:
Initialize next release cycle for Boost (#23341)

Co-authored-by: Mark George <thingalon@gmail.com>
Co-authored-by: Jeremy Herve <jeremy@jeremy.hu>

kangzj committed:
Search package: dashboard use local gridicon (#22915)

Co-authored-by: Kevin L <40267301+a8ck3n@users.noreply.github.com>

BogdanUngureanu commented:
Can you merge master again into this branch? The diff couldn't be applied cleanly on the sandbox (even after running Jetpack fusion again).

anomiex committed:
eslint-changed: Remove unneeded dependency on `@babel/preset-react` (#23338)

This was added in error in #22360. The actual fix was to add the missing
dependency where it was needed, which was done in #22359.

rcrdortiz committed:
Merge branch 'fix/show-premium-content-after-subscribe' of github.com:Automattic/jetpack into fix/show-premium-content-after-subscribe

rcrdortiz commented:
Moved to [23347](https://github.com/Automattic/jetpack/pull/23347)
